src/wazuh_modules/syscollector/src/syscollectorImp.cpp

```diff
@@ -1657,10 +1659,48 @@ void Syscollector::push(const std::string& data)
         auto rawData{data};
         Utils::replaceFirst(rawData, "dbsync ", "");
         const auto buff{reinterpret_cast<const uint8_t*>(rawData.c_str())};
+        bool pushMessage = true;
+        std::string component;
+        std::string topic;
+        time_t id;
+        bool parseSuccesfull = true;
 
         try
         {
-            m_spRsync->pushMessage(std::vector<uint8_t> {buff, buff + rawData.size()});
+            component = Utils::splitIndex(rawData, ' ', 0);
+            topic = Utils::splitIndex(rawData, ' ', 1);
+            const nlohmann::json syncInfoJson = nlohmann::json::parse(Utils::splitIndex(rawData, ' ', 2));
+            id = std::stol(syncInfoJson.at("id").dump());
+        }
+        catch (...)
+        {
+            parseSuccesfull = false;
+        }
+
+        try
+        {
+            if (parseSuccesfull)
+            {
+                const auto itComponent = m_lastSyncMsg.find(component);
+
+                if (itComponent == m_lastSyncMsg.end())
+                {
+                    m_lastSyncMsg[component] = id;
+                }
+
+                if (m_intervalValue + m_scanDuration <= id - m_lastSyncMsg[component] &&
+                        (2 * (m_intervalValue + m_scanDuration) > id - m_lastSyncMsg[component]))
+                {
+                    m_logFunction(LOG_DEBUG_VERBOSE, "Synchronization in progress. Discarded message: " + rawData);
+                    pushMessage = false;
+                    m_lastSyncMsg[component] = id;
+                }
```

Comment:
We can move all this logic to the first try-catch block to:
- make the `std::string component;` and `time_t id;` variables constant
- Avoid the use of `bool parseSuccesfull = true;`

Reply:
Much better